samples/settings/prj.conf

```diff
@@ -3,17 +3,6 @@ CONFIG_GOLIOTH_SAMPLES_COMMON=y
 # Application
 CONFIG_LOG_BACKEND_GOLIOTH=y
 
-CONFIG_REBOOT=y
-
-CONFIG_FLASH=y
-CONFIG_FLASH_MAP=y
-CONFIG_NVS=y
-
-CONFIG_SETTINGS=y
-CONFIG_SETTINGS_RUNTIME=y
-CONFIG_GOLIOTH_SAMPLE_SETTINGS_AUTOLOAD=y
-CONFIG_GOLIOTH_SAMPLE_SETTINGS_SHELL=y
-
```

Comment:
> enable storing PSK-ID/PSK in settings subsystem, so that the DFU sample can be flashed on multiple devices as is and then do a DFU of the fleet

Yes, this is how precompiled binaries for the zephyr-training, thingy91demo, and all the reference designs currently work. We're using PSK auth but it's not hardcoded into the firmware.

Reply:
Hardcoded PSKs are just the default. As part of this PR, there's a new overlay config file in `samples/common` that can enable runtime PSKs via the settings subsystem for _any_ of our samples, e.g.:
`west build -p -b esp32 samples/dfu -- -DOVERLAY_CONFIG="../common/runtime_psk.conf"`

So this should give us the best of both worlds, while maintaining consistency among our samples. All the samples use hardcoded PSKs by default, and all can be changed to use runtime PSKs and the settings shell with a single build argument. This is documented in the READMEs for each of the samples now, e.g.: https://github.com/golioth/golioth-zephyr-sdk/pull/406/files#diff-40c1e6da1ab07e4c146dae9c4f8ecf4c2dd4afdf703d11f21d10dc6b2e4898c3R46